server.go

```diff
@@ -122,30 +120,39 @@ func (s *Server) Close() error {
 	return err
 }
 
-func (s *Server) readPacketConn(p PacketConnConfig, am *allocation.Manager) {
-	s.readLoop(p.PacketConn, am)
+func (s *Server) readListener(l net.Listener, am *allocation.Manager) {
+	for {
+		conn, err := l.Accept()
+		if err != nil {
+			s.log.Debugf("Failed to accept: %s", err)
+			return
+		}
+
+		go func() {
+			s.readLoop(NewSTUNConn(conn), am)
 
-	if err := am.Close(); err != nil {
-		s.log.Errorf("Failed to close AllocationManager: %s", err)
+			if err := conn.Close(); err != nil {
+				s.log.Errorf("failed to close conn: %s", err)
```

Comment:
I'm not sure we want to report all these `"failed to close conn` errors: most of the time the connection is already closed by the time we get here, no? Won't this yield lots of useless errors?

Reply:
We are now only showing an error if its not an `net.ErrClosed`